src/routes/podcasts/[id].svelte

```diff
@@ -1,37 +1,15 @@
 <script lang="ts" context="module">
-  import type { Load } from '@sveltejs/kit';
-  import { podcastGet } from '$lib/entities/podcast';
-
-  export const load: Load = async ({ params }) => {
-    try {
-      const podcast = await podcastGet(parseInt(params.id, 10));
-      return { props: { podcast } };
-    } catch (e) {
-      console.error(e);
-      return { props: { podcast: undefined } };
-    }
-  };
+  export { loadPodcastFromURL as load } from '$lib/pages/podcast-episodes';
 </script>
 
 <script lang="ts">
   import { PodcastEpisodesPage } from '$lib/pages/podcast-episodes';
+  import { addPodcastToStore } from '$lib/entities/podcast';
   import type { Podcast } from '$lib/shared/api';
-  import { podcasts as podcastStore } from '$lib/entities/podcast';
 
-  export let podcast: Podcast | undefined;
+  export let podcast: Podcast;
 
-  if (podcast) {
-    podcastStore.update((podcasts) => {
-      if (!podcasts.some((p) => p.id === podcast!.id)) {
-        podcasts.push(podcast!);
-      }
-      return podcasts;
-    });
-  }
+  addPodcastToStore(podcast);
```

Comment:
Question: what does adding a podcast to the list of podcasts do?

Reply:
It caches that podcast's metadata to avoid downloading it again later